src/NServiceBus.Core/MessageInterfaces/MessageMapper/Reflection/MessageMapper.cs

```diff
@@ -250,6 +259,8 @@ static string GetTypeName(Type t)
 
         readonly object messageInitializationLock = new object();
 
+        HashSet<Type> initializedTypes = new HashSet<Type>();
```

Comment:
@andreasohlund i thought on the call we had dicussed consolidating the existing dictionaries, not adding a new lookup?